### PR TITLE
Add a v3 migration page

### PR DESCRIPTION
**Pull request by skurnevich**
Describe your pull request here:

List the file(s) included in this PR:

After creating the PR, follow the instructions in the comments.

### CODE REVIEW

File: docs/whats-new/zowe-v3-migration.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+The service ID for gateway zosmf has changed to **ibmzosmf**. If you are using zosmf as your auth service, this needs to be updated.
+
+```yaml
+components:
+  gateway:
+    apiml:
+      security:
+        auth:
+          zosmf:
+             jwtAutoconfiguration: jwt
```
Comment (1000TurquoisePogs) [inline docs/whats-new/zowe-v3-migration.md:38]: also, "auto" has been removed.
a user must either choose "jwt" (default) or "ltpa"
both rocket and ibm are having trouble using jwt, and are using "ltpa".

File: docs/whats-new/zowe-v3-migration.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+- The following network changes are needed for added or removed servers:
+
+| Component name | Change | Default Port | Default Jobname | Details |
+|----------------|--------|--------------|-----------------|---------|
+| zaas | Added | 7558 | ZWE1AZ | This component is responsible for authentication and is now required when using the API Mediation Layer |
+| metrics-service | Removed | 7551 | ZWE1MS | This service has been deprecated and removed. Currently, no replacement is available. The Open Telemetry standard will be implemented later, which will serve as a replacement |
+| jobs-api | Removed | 7558 | ZWE1EJ | This component was deprecated in Zowe v2 and is now removed. Ensure that you switch to using equivalent z/OSMF endpoints |
+| files-api | Removed | 7559 | ZWE1EF | This component was deprecated in Zowe v2 and is now removed. Ensure that you switch to using equivalent z/OSMF endpoints |
+| cloud-gateway | Removed | 7563 | ZWE1CG | The cloud-gateway has replaced the main gateway component in v3 and is now accessible on 7554 as ZWE1AG |
```
Comment (1000TurquoisePogs) [inline docs/whats-new/zowe-v3-migration.md:84]: this detail about cloud-gateway replaces gateway yet takes on the same name and port as gateway is perhaps too much technical information for users - I shared this document with users and they didn't know what it meant.

Comment (balhar-jakub) [reply docs/whats-new/zowe-v3-migration.md:84]: Then we should just mention that it is removed here

Comment (skurnevich) [reply docs/whats-new/zowe-v3-migration.md]: Reverted it to the previous phrase "The cloud-gateway has been removed as a standalone component and merged into the gateway", looks ok?

**Review by janan07 — APPROVED** [review]
Reviewed and edited by doc squad

**Pull request merged**